### ISSUE TITLE
Linear probing training script

### ISSUE THREAD

**Issue by zelaki** [sihyun-yu/REPA#32]
Hello and thank you for your fascinating work,

I want to reproduce your linear probing results.
Could you let me know if you are planning to make the training script public?

Thank you in advance,
Thodoris

**Comment by aSleepyTree** [comment]
+1

**Comment by dekrt** [comment]
+1

**Comment by sihyun-yu** [comment]
Sorry for the late reply. Will try to make the code public after refactoring :( 
Here are some details: we did not inject any class labels (to prevent leakages). We use input vectors as averaged hidden state across patch axis (so each vector becomes a 1D vector where dimension = hidden dimension of DiT). For the linear classifier, we used the following architecture with a batchnorm layer:

```python
class LinearModel(nn.Module):
    def __init__(self, input_size, num_classes):
        super(LinearModel, self).__init__()
        self.norm1 = nn.BatchNorm1d(hidden_dim)
        self.linear = nn.Linear(input_size, num_classes)

    def forward(self, x):
        x = self.norm1(x)
        return self.linear(x)
```
